src/ploigos_step_runner/utils/xml.py

```diff
@@ -87,3 +91,100 @@ def get_xml_element_by_path(xml_file_path, xpath, default_namespace=None, xml_na
         return xml_file.find(xpath)
 
     return xml_file.find(xpath, namespaces)
+
+def aggregate_xml_element_attribute_values(xml_file_path, xml_element, attribs): # pylint: disable=too-many-branches
+    """
+    Function to parse a XML file looking for a specific element and obtaining its attributes.
+
+    If element is not found in file the file is ignored.
+
+    If attribute is not found in element the file is ignored.
+
+    Parameters
+    ----------
+    xml_file_path: str
+        Path to directory containing XML files
+    xml_element: str
+        XML element being searched for
+    attribs: list
+        List of attributes being searched for in the XML element
+
+    Returns
+    --------
+    report_results: dict
+        A dictionary of attribute values.
+
+    Raises
+    ------
+    ValueError
+        If XML element is not found in file(s)
+
+    """
+
+    report_results = {}
+
+    #Search file path for XML files
+
+    if os.path.isdir(xml_file_path):
+        xml_files = glob.glob(xml_file_path + '/*.xml', recursive=False)
+    elif os.path.isfile(xml_file_path):
+        xml_files = [xml_file_path]
+
+    with open(xml_files[0], "r") as xml_file:
+        print("line: " + xml_file.read())
+
+    for attrib in attribs:
+        report_results[attrib] = []
+
+    #Iterate over XML files
+    for file in xml_files:
+        try:
+            pom_element = get_xml_element(file, xml_element)
+            #Add attributes to dictionary
+            for attrib in attribs:
+                attribute_value = pom_element.attrib[attrib]
+                # aggregate attribute accross multiple files
+                report_results[attrib].append(attribute_value)
+        except ValueError:
+            #Return empty dict if an attribute is missing
+            return {}
+
+    #Loop to check if all values aggregated are integers, floats, strings, or a mix
+    #of all three.
+    for attrib in attribs:
+        only_numerics = True
+        only_floats = True
+        #Iteration to determine what pattern all values follow.
+        for value in report_results[attrib]:
+            #isnumeric() returns True if the value contains numbers
+            #but not a decimal point. If this returns
+            #true then continue because the value would be
+            #a valid float anyway
+            if not only_numerics and not only_floats:
```

Comment:
I just realized - since we prioritize numeric first, then floats, we don't even need this check. We can just add the `break` after line 171, because once only_floats is false, only_integers is guaranteed to be false as well.

Reply:
Good point. Wish I had thought of that.

Reply:
[Fixed](https://github.com/ploigos/ploigos-step-runner/pull/155/commits/3ee3467e421ba1951b632c7811b6e143ecc667ce)